src/getgb2.f

```diff
@@ -1,291 +1,235 @@
+C>    contains an entry (in the order specified) of Product Defintion
+C>    Template 4.N. This element is actually a pointer to an array
+C>    that holds the data.
+C>    - gfld\%ipdtlen Number of elements in gfld\%ipdtmpl. i.e. number of
+C>    entries in Product Defintion Template 4.N (N=gfdl\%ipdtnum).
+C>    - gfld\%coord_list Real array containing floating point values
+C>    intended to document the vertical discretisation associated to
+C>    model data on hybrid coordinate vertical levels.(part of Section 4)
+C>    This element is actually a pointer to an array
+C>    that holds the data.
+C>    - gfld\%num_coord number of values in array gfld\%coord_list.
+C>    - gfld\%ndpts Number of data points unpacked and returned.
+C>    Note that this number may be different from the value of
+C>    - gfld\%ngrdpts if a logical bitmap is in use with grid points
+C>    that are being masked out.
+C>    - gfld\%idrtnum Data Representation Template Number (see Code Table 5.0)
+C>    - gfld\%idrtmpl Contains the data values for the specified Data
+C>    Representation Template (N=gfld\%idrtnum). Each element of this
+C>    integer array contains an entry (in the order specified) of
+C>    Product Defintion Template 5.N. This element is actually a
+C>    pointer to an array that holds the data.
+C>    - gfld\%idrtlen Number of elements in gfld\%idrtmpl. i.e. number
+C>    of entries in Data Representation Template 5.N (N=gfld\%idrtnum).
+C>    - gfld\%unpacked logical value indicating whether the bitmap and
+C>    data values were unpacked. If false, gfld\%bmap and gfld\%fld
+C>    pointers are nullified.
+C>    - gfld\%expanded Logical value indicating whether the data field
+C>    was expanded to the grid in the case where a bit-map is present.
+C>    If true, the data points in gfld\%fld match the grid points and
+C>    zeros were inserted at grid points where data was bit-mapped out.
+C>    If false, the data values in gfld\%fld were not expanded to the
+C>    grid and are just a consecutive array of data points corresponding
+C>    to each value of "1" in gfld\%bmap.
+C>    - gfld\%ibmap Bitmap indicator (see Code Table 6.0)
+C>    0 bitmap applies and is included in Section 6.
+C>    1-253 Predefined bitmap applies
+C>    254 Previously defined bitmap applies to this field
+C>    255 Bit map does not apply to this product.
+C>    - gfld\%bmap Logical*1 array containing decoded bitmap, if ibmap=0
+C>    or ibap=254. Otherwise nullified. This element is actually a
+C>    pointer to an array that holds the data.
+C>    - gfld\%fld Array of gfld\%ndpts unpacked data points. This element
+C>    is actually a pointer to an array that holds the data.
+C>    @param[out] IRET integer return code
+C>    - 0 all ok
+C>    - 96 error reading index
+C>    - 97 error reading grib file
+C>    - 99 request not found
+C>    - other gf_getfld grib2 unpacker return code
+C>
+C>    Subprograms called:
```

Comment:
You can remove the section "subprograms called" everywhere you see it. Doxygen has a neat feature we can turn on which automatically provides a list of all subprograms called, which we then don't have to maintain. (If we are maintaining it, it inevitably will become wrong over time, whereas doxygen figures it out automatically each time and is always correct).

Reply:
Thanks for the knowledge!